R/calculate_recommended_interventions.R

```diff
@@ -302,26 +372,275 @@ calculate_recommended_interventions <- function(
+    if (include_center_effects) {
+      if (input_data_structure == "center_level") {
+        center_sizes <- tapply(
+          data$center_sample_size,
+          data$center,
+          function(x) x[1]
+        )
+        total_sample_size <- sum(center_sizes)
+      } else if (input_data_structure == "individual_level") {
+        total_sample_size <- sum(as.numeric(table(data$center)))
+        center_sizes <- as.numeric(table(data$center))
+      }
+      center_weights_for_outcome_goal <- center_sizes / total_sample_size
+    } else {
+      center_weights_for_outcome_goal <- 1
+    }
+  }
+
+  # check if values of center_weights_for_outcome_goal sum up to 1
+  if (!is.null(center_weights_for_outcome_goal)) {
+    if (sum(center_weights_for_outcome_goal) != 1) {
+      stop(paste(
+        "values in center_weights_for_outcome_goal must",
+        "sum up to 1."
+      ))
+    }
+  }
+
+  # check if include_time_effects is logical
+  if (!is.logical(include_time_effects)) {
+    stop(paste("'include_time_effects' must be a boolean."))
+  }
+
+  # if include_time_effects is set to TRUE
+  if (include_time_effects) {
+    # the input data must have a 'period' column
+    if (!"period" %in% names(data)) {
+      stop(paste(
+        "To include fixed time effects in the outcome model,",
+        "please make sure that the input data has a 'period'",
+        "column to identify time period."
+      ))
+    }
+    # the time_effect_optimization_value must be provided, one numeric number
+    if (is.null(time_effect_optimization_value) ||
+      !is.numeric(time_effect_optimization_value) ||
+      length(time_effect_optimization_value) != 1) {
+      stop(paste(
+        "'include_time_effects' is set to TRUE,",
+        "'time_effect_optimization_value' must be provided,",
+        "and it must be a numeric value."
+      ))
+    }
+    # if "period" column of the input data is not a factor,
+    # change it to factor and print a warning message
+    if (!is.factor(data$period)) {
+      data$period <- as.factor(data$period)
+      message(paste(
+        "'period' column is not a factor type. To ensure",
+        "the correct model fit, it has been converted",
+        "to the factor type."
+      ))
+    }
+  }
+
+  # if include_interaction_terms is set to TRUE
+  if (include_interaction_terms) {
+    # check if interaction terms are included in intervention_components
+    # and follow the correct naming scheme
+    if (!any(grepl(":", intervention_components, fixed = TRUE))) {
+      stop(paste(
+        "Please make sure the interaction terms are included",
+        "as part of the 'intervention_components', and the interaciton terms",
+        "need to have the format of 'X1:X5' "
+      ))
+    }
+
+    # if so, check if main_components is null
+    if (is.null(main_components)) {
+      stop(paste(
+        "'main_components' must be defined if interaction terms are",
+        "included in 'intervention_components'."
```

Comment:
Can we add this here to clarify that 'main_components' must be a character vector? I know you have this error handle right underneath, but just to be clear and precise if the error was this instead.

Reply:
fixed